Search, One-page forms and User mgmt section review (#10)
ezsystems/commerce-documentation: docs/guide/user_management/customer_center/customer_center_functions/customer_center_user_actions.md

# Customer center user actions

## Edit, add and request new user

There are default forms prepared for the actions of editing users that can be partially modified in the configuration.

There are two types of form fields:

- static fields - cannot be removed from the form, some settings (like validation or name) can be configured.
- dynamic fields - can be removed or added, are read/stored in the content if corresponding content Field identifier exists in the User Content Type and can be sent to ERP (request new user).

### Form field configuration

> You can change the [form field type](http://symfony.com/doc/current/reference/forms/types.html).

Reviewer: ```suggestion
You can change the [form field type.](http://symfony.com/doc/3.4/reference/forms/types.html)
```

Except for `submit`, all field values are stored in the User Content item and can be edited in the Back Office.

## Setting up a new field

All custom fields must be defined in the `dynamic` section of the form's attributes.
These fields have to be added to the User Content Type in the Back Office and the field identifier must be the same as the key in the configuration.

For a detailed description, see [Extending the Customer center forms](../customer_center_cookbook.md#extending-the-customer-center-forms).

## Feedback messages

Every form defines a message for successful submission and for an error that can occur during form data processing.
The configuration keys are `validMessage` and `invalidMessage`, respectively.
It's best practice to define identifiers for [text modules](../../../translations/translations.md)
and edit the feedback content in the Back Office.
The content of `invalidMessage` should contain correction suggestions.

## Initial values

A form can define a service in the `initialFormValuesService` key that pre-fills the form fields with specific values.
These values can be read from an external data source or from content.

> The configuration value must a valid service ID and the service's class must implement [`InitialFormValuesServiceInterface`](../customer_center_api/initialformvaluesserviceinterface.md).

Reviewer: ```suggestion
The configuration value must be a valid service ID and the service's class must implement [`InitialFormValuesServiceInterface`](../customer_center_api/initialformvaluesserviceinterface.md).
```
OR
```suggestion
The configuration value must have a valid service ID and the service's class must implement [`InitialFormValuesServiceInterface`](../customer_center_api/initialformvaluesserviceinterface.md).
```

## Form processors

The form processors are services that perform special actions after the form is submitted.

> They are executed sequentially according the order of definition in the configuration.

Reviewer: ```suggestion
They are executed sequentially according to the order of definition in the configuration.
```
